packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -43,7 +44,7 @@ interface CommonProps
    * The underlying input element type
    * @default text
    */
-  type?: "text" | "email" | "url" | "tel";
+  type?: "text" | "email" | "url" | "tel" | "search" | "password";
```

Comment:
Ideally these types would no longer be available here to reduce confusion/duplication across our total API surface area. I think we might still allow for the underlying technique you're after in this PR by using a `@ts-expect-error` in Password and Search components to use the prop we want and assign the underlying input type irregardless of type warnings.

Reply:
good idea, i'll try that